File: src/main/java/com/giant/common/config/security/type/Role.java
package com.giant.common.config.security.type;

import lombok.Getter;

import java.util.Arrays;

@Getter
public enum Role {
    ADMIN("1", "ROLE_ADMIN"),
    USER("2", "ROLE_USER"),
    LEAVING("3", "ROLE_LEAVING"),
    GUEST("4", "ROLE_GUEST");

    private final String code;
    private final String authority;

    Role(String code, String authority) {
        this.code = code;
        this.authority = authority;
    }

    public static Role fromCode(String code) {
        return Arrays.stream(Role.values())
                .filter(role -> role.code.equals(code))
                .findFirst()
                .orElse(GUEST);
    }
}

File: src/main/kotlin/com/giant/common/config/security/type/Role.kt
package com.giant.common.config.security.type

enum class Role(val code: String, val authority: String) {
    ADMIN("ADMIN", "ROLE_ADMIN"),
    USER("USER", "ROLE_USER"),
    LEFT("LEFT", "ROLE_LEFT"),
    GUEST("GUEST", "ROLE_GUEST");

    companion object {
        fun fromCode(code: String?): Role =
            Role.entries.find { it.code == code } ?: GUEST
    }
}